--- /home/andy/.vim/syntax/2html.vim	2009-10-10 15:32:39.000000000 +0000
+++ /usr/share/vim/vim72/syntax/2html.vim	2009-06-16 06:24:31.000000000 +0000
@@ -192,6 +192,7 @@
 let s:old_report = &report
 let s:old_search = @/
 set notitle noicon
+setlocal et
 set report=1000000
 
 " Split window to create a buffer with the HTML file.
@@ -244,7 +245,7 @@
 " HTML header, with the title and generator ;-). Left free space for the CSS,
 " to be filled at the end.
 exe "normal! a<html>\n\e"
-exe "normal! a<head>\n<title>" . &titlestring . "</title>\n\e"
+exe "normal! a<head>\n<title>" . expand("%:p:~") . "</title>\n\e"
 exe "normal! a<meta name=\"Generator\" content=\"Vim/" . v:version/100 . "." . v:version %100 . '"' . s:tag_close . "\n\e"
 if s:html_encoding != ""
   exe "normal! a<meta http-equiv=\"content-type\" content=\"text/html; charset=" . s:html_encoding . '"' . s:tag_close . "\n\e"
@@ -392,16 +393,12 @@
 
       " Expand tabs
       let s:expandedtab = strpart(s:line, s:startcol - 1, s:col - s:startcol)
-      if &et
-        let idx = stridx(s:expandedtab, "\t")
-        while idx >= 0
-          let i = &ts - ((idx + s:startcol - 1) % &ts)
-          let s:expandedtab = substitute(s:expandedtab, '\t', repeat(' ', i), '')
-          let idx = stridx(s:expandedtab, "\t")
-        endwhile
-      else
-        setlocal isprint+=9
-      endif
+      let idx = stridx(s:expandedtab, "\t")
+      while idx >= 0
+	let i = &ts - ((idx + s:startcol - 1) % &ts)
+	let s:expandedtab = substitute(s:expandedtab, '\t', repeat(' ', i), '')
+	let idx = stridx(s:expandedtab, "\t")
+      endwhile
 
       " Output the text with the same synID, with class set to {s:id_name}
       let s:id = synIDtrans(s:id)
